### ISSUE TITLE
container pipeline fail points to possible security risk when pulling npm dependencies

### ISSUE THREAD

**Issue by zem** [xwiki-labs/cryptpad#668]
**Describe the bug**

I have encountered a minor but dangerous problem when I was trying to build a cryptpad container on my gitlab instance https://build.conesphere.com/conesphere/podman/hub/cryptpad 

```
INFO[0056] args: [-c npm install --production     && npm install -g bower     && bower install --allow-root] 
INFO[0056] Running: [/bin/sh -c npm install --production     && npm install -g bower     && bower install --allow-root] 
npm ERR! Error while executing:
npm ERR! /usr/bin/git ls-remote -h -t git://github.com/dchest/tweetnacl-js.git
npm ERR! 
npm ERR! fatal: unable to connect to github.com:
npm ERR! github.com[0: 140.82.121.4]: errno=Connection timed out
npm ERR! 
npm ERR! 
npm ERR! exited with error code: 128
npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2020-12-23T22_09_38_061Z-debug.log
error building image: error building stage: failed to execute command: waiting for process to exit: exit status 1
ERROR: Job failed: exit code 1
```

The reason why that pipeline failed was rather simple: The build runner is not allowed to pull any software from insecure (http!!) resources. 

I did forbid http and smtp on my gitlab-runner cluster for good reasons, like preventing software from completely unverified sources to get into production! 

The error can be of course circumvented by doing: 

```
git config --global url."https://".insteadOf git://
```
However this leaves 2 major problems: 

1. Why is cryptpad npm sourcing in stuff via http (git)?
2. (Why is git on this debian based container not using https by default for any git:// sources as documented?) 

As the version of git is not neccessarily the newest, I would sugest to go for an exchange of any git:// url in the npm packages sources with https:// which seems is the safe option here. 

**Where did it happen?**
Did the issue occur on CryptPad.fr or an instance hosted by a third-party?  
If on another instance, please provide its full URL.

**To Reproduce**
1. Block http
2. run npm install --production     && npm install -g bower     && bower install --allow-root

**Expected behavior**
Installation should work...

**Comment by ansuz** [comment]
It looks like the offending URI is included via [this dependency](https://github.com/xwiki-labs/cryptpad/blob/main/package-lock.json#L183-L196). As I recall that was hastily copied from our bower dependency list so that we could easily test some client-side components from the command line and it isn't used by the nodejs process at all.

Thanks for pointing this out. I'll try to fix it soon.

**Comment by ansuz** [comment]
This was merged to our `main` branch for our 4.0.0 release. We expect to tag later today.
